dtm0/clk_src.h

```diff
@@ -0,0 +1,167 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#pragma once
+#ifndef __MOTR_DTM_CLK_SRC_H__
+#define __MOTR_DTM_CLK_SRC_H__
+
+#include "xcode/xcode.h" /* xcode attrs */
+#include "lib/time.h" /* m0_time_t */
+#include "lib/mutex.h" /* m0_mutex */
+
+/* DTM clock source overview
+ * -------------------------
+ *
+ *   Clock source module is designed to be used by DTM0. At this point,
+ * only one type of clock is supported -- physical clock. Such a clock
+ * provides a way to enforce total ordering in a system with synchronised
+ * clocks. However, the interface supports an epoch-based clock source as well.
+ *   See "A.clock.sync" in the DTM0 HLD for additional details.
```

Comment:
Thanks for pointing at this module; I wasn't aware of its existence. I glanced through the code, and it looks like the the thing I imagined when was writing this paragraph. If we are going to enter the field of hybrid clocks (or pure logical clocks) then we definitely should consider this lclock module. I have pointed out that this module exists in a comment at the end (https://github.com/Seagate/cortx-motr/pull/561/commits/0ec44d05c4f76f18b53b4e4b77a0073561ae2097). I hope it will be enough at this point.